src/services/validationService.ts

```diff
@@ -0,0 +1,164 @@
+import {
+  ICommonRules,
+  IRules,
+  IValidationResponse,
+  INativeQuery,
+  IDataSourceRules,
+  IQueryTypeRules,
+  IQuery,
+  ISqlQuery,
+} from "../models";
+import constants from "../resources/constants.json";
+import { isEmpty, isUndefined } from "lodash";
+import moment from "moment";
+import fs from "fs";
+import Ajv from "ajv";
+
+export class ValidationService {
+  public static validateNativeQuery(
+    query: INativeQuery,
+    commonLimits: ICommonRules,
+    dataSourceLimits: IDataSourceRules
+  ): IValidationResponse {
+    let nativeQuerySchema = JSON.parse(
+      fs.readFileSync(
+        process.cwd() + "/src/configs/nativeQuerySchema.json",
+        "utf8"
+      )
+    );
+    let ajv = new Ajv();
+    let isValidQuery = ajv.validate(
+      nativeQuerySchema,
+      JSON.parse(JSON.stringify(query))
+    );
+    if (isValidQuery) {
+      if (!isEmpty(dataSourceLimits) && !isUndefined(dataSourceLimits)) {
+        this.determineQueryLimits(query, commonLimits);
+        try {
+          return this.validateQueryRules(
+            query,
```

Comment:
Why do we need the try catch block ? is `validateQueryRules` query not returning the validationStatus object ?